File: VB/WinWebSolution.Module/SecuritySystemHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Utils
Imports DevExpress.Persistent.Base.Security

Namespace Demo
	Public Class SecuritySystemHelper
		Public Shared Function IsUserInRole(ByVal userWithRoles As IUserWithRoles, ByVal roleName As String) As Boolean
			Guard.ArgumentNotNull(userWithRoles, "userWithRoles")
			For Each role As IRole In userWithRoles.Roles
				If role.Name = roleName Then
					Return True
				End If
			Next role
			Return False
		End Function
		Public Shared Function IsUserInRole(ByVal roleName As String) As Boolean
			Return IsUserInRole(TryCast(SecuritySystem.CurrentUser, IUserWithRoles), roleName)
		End Function
		Public Shared Function IsUserAdministrator(ByVal simpleUser As ISimpleUser) As Boolean
			Guard.ArgumentNotNull(simpleUser, "simpleUser")
			Return simpleUser.IsAdministrator
		End Function
	End Class
End Namespace

File: CS/WinWebSolution.Module/SecuritySystemHelper.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Utils;
using DevExpress.Persistent.Base.Security;

namespace Demo {
    public class SecuritySystemHelper {
        public static bool IsUserInRole(IUserWithRoles userWithRoles, string roleName) {
            Guard.ArgumentNotNull(userWithRoles, "userWithRoles");
            foreach (IRole role in userWithRoles.Roles)
                if (role.Name == roleName)
                    return true;
            return false;
        }
        public static bool IsUserInRole(string roleName) {
            return IsUserInRole(SecuritySystem.CurrentUser as IUserWithRoles, roleName);
        }
        public static bool IsUserAdministrator(ISimpleUser simpleUser) {
            Guard.ArgumentNotNull(simpleUser, "simpleUser");
            return simpleUser.IsAdministrator;
        }
    }
}